lib/restforce/db/field_processor.rb

```diff
@@ -21,20 +25,35 @@ def self.reset
         @field_cache = {}
       end
 
+      # Public: Get a list of valid fields for a specific action from the passed
+      # list of proposed fields. Allows access to related object fields on a
+      # read-only basis.
+      #
+      # sobject_type - A String name of an SObject Type in Salesforce.
+      # attributes   - A Hash with keys corresponding to Salesforce field names.
+      # action       - A Symbol reflecting the action to perform. Accepted
+      #                values are :read, :create, and :update.
+      #
+      # Returns a Hash.
+      def available_fields(sobject_type, fields, action = :read)
+        fields.select do |field|
+          available?(sobject_type, field, action) || (
```

Comment:
This formatting reads oddly to me; perhaps re-format to be on one line _or_ introduce a local or new private method.